sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientException.java

```diff
@@ -248,6 +250,15 @@ CosmosClientException setCosmosResponseDiagnostics(CosmosResponseDiagnostics cos
         return this;
     }
 
+    public RequestTimeline getRequestTimeline() {
+        return this.requestTimeline;
+    }
+
+    public CosmosClientException setRequestTimeline(RequestTimeline requestTimeline) {
```

Comment:
setter should not be public. if setter needed for internal implementation we should so similar to `CosmosClientException::setCosmosResponseDiagnostics(.)

Reply:
fixed on my local branch. good catch. thanks.